src/main/java/edu/byu/cs/util/DataUtils.java

```diff
@@ -0,0 +1,28 @@
+package edu.byu.cs.util;
+
+import edu.byu.cs.dataAccess.DaoService;
+import edu.byu.cs.model.Phase;
+import edu.byu.cs.model.Submission;
+
+import java.util.Collection;
+
+public class DataUtils {
+
+    /**
+     * Returns the student's first passing submission for the given phase
+     *
+     * @param netID the netID of the student
+     * @param phase the phase they passed off (or didn't)
+     * @return the Submission object, or null
+     */
+    public static Submission getFirstPassingSubmission(String netID, Phase phase) {
+        Collection<Submission> submissions = DaoService.getSubmissionDao().getSubmissionsForPhase(netID, phase);
+        Submission firstPassing = null;
+        int earliest = Integer.MAX_VALUE;
+        for (Submission s : submissions) {
+            if (!s.passed()) continue;
+            if (s.timestamp().getEpochSecond() < earliest) firstPassing = s;
+        }
+        return firstPassing;
+    }
```

Comment:
I think this should be a Dao method instead. I know you didn't write this initially, but I don't think it belongs in a class called `DateUtils`. If you'd rather not change it in this PR, could you open a new issue for it? #cleancode 😃

Reply:
Done